sequencing/ATACSeq/Rscripts/collateSexChecks.r

```diff
@@ -1,34 +1,65 @@
+##                                                                                           ||
+## DESCRIPTION: This scripts collates results from calling peaks in sex chromosomes and      ||
+## uses this to check the assigned sex of samples.                                           ||
+##                                                                                           ||
+## INPUTS:                                                                                   ||
+## - <config file> : R config file for project                                               ||
+##                                                                                           ||
+## OUTPUTS:                                                                                  ||
+## - sexPredictions.csv, sexPredictions.pdf                                                  ||
+##                                                                                           ||
+## REQUIRES:                                                                                 ||
+## - R/4.2.1-foss-2022a, libraries: ggpubr, tidyverse, conflicted,ggplot2                    ||
+## ==========================================================================================##
+
 args<-commandArgs(trailingOnly=TRUE)
-dataPath<-args[1]
+configFile<-args[1]
+source(configFile)
 
 library(ggplot2)
 library(ggpubr)
 library(tidyverse)
-
-
+library(conflicted)
+conflict_prefer("filter", "dplyr")
+conflict_prefer("lag", "dplyr")
 ## reads in peaks annotated to XIST and FIRRE
-sampleSheet<-read.csv(file.path(dataPath, "0_metadata", "sampleSheet.csv"), stringsAsFactors = FALSE)
+sampleSheet<-read.csv(file.path(dir, "/0_metadata", "sampleSheet.csv"), stringsAsFactors = FALSE,colClasses="character")
 
-xPeakFile<-file.path(dataPath, "5_countPeaks", "MACS","ShiftedTagAlign", "sexChr", "chrX.peakcounts.txt")
-yPeakFile<-file.path(dataPath, "5_countPeaks", "MACS","ShiftedTagAlign","sexChr", "chrY.peakcounts.txt")
-xPeaks<-read.table(xPeakFile, stringsAsFactors = FALSE)
-yPeaks<-read.table(yPeakFile, stringsAsFactors = FALSE)
+#For Shifted Tag Align mode
+xPeakFile<-file.path(dir, "/5_countPeaks", "ShiftedTagAlign", "sexChr", "chrX.peakcounts.txt")
+yPeakFile<-file.path(dir, "/5_countPeaks", "ShiftedTagAlign","sexChr", "chrY.peakcounts.txt")
 
+xPeaks<-read.table(xPeakFile, stringsAsFactors = FALSE)
+yPeaks<-read.table(yPeakFile, stringsAsFactors = FALSE,fill = TRUE)
 
 xPeaks$V18<-basename(xPeaks$V18)
-yPeaks$V10<-basename(yPeaks$V10)
+yPeaks$V11<-basename(yPeaks$V11)
+
 xPeaksWide = xPeaks %>% 
 	select(V4, V17, V18) %>% 
   spread(V4, V17)
 
 yPeaksWide = yPeaks %>% 
-	select(V4, V10, V11) %>% 
-  spread(V4, V11)
-  
-xPeaksWide$YTot<-rowSums(yPeaksWide[,-1])
-xPeaksWide$sex<-sampleSheet$gender[match(gsub(".chrX.tn5.tagAlign.gz", "", xPeaksWide$V18), sampleSheet$sampleID)]
+	select(V4, V10, V11) %>%
+ spread(V4, V10)
```

Comment:
I realise that there is a lot of columns in xPeaks and yPeaks and this will be a pain, but would it be possible to give names to the columns that are being manipulated here? 

You could probably get away with just giving names to the columns that are being manipulated (`V4`,`V10`,`V11`,*etc*). This should make the code more readable in the future.